zebra-chain/src/block/height.rs

```diff
@@ -65,6 +65,29 @@ impl Height {
     /// previous to Nu5 and in non-coinbase transactions from Nu5 activation
     /// height and above.
     pub const MAX_EXPIRY_HEIGHT: Height = Height(499_999_999);
+
+    /// Returns the next [`Height`].
+    ///
+    /// # Panics
+    ///
+    /// - If the current height is at its maximum.
+    pub fn next(self) -> Self {
+        (self + 1).expect("Height should not be at its maximum.")
+    }
+
+    /// Returns the previous [`Height`].
+    ///
+    /// # Panics
+    ///
+    /// - If the current height is at its minimum.
+    pub fn previous(self) -> Self {
+        (self - 1).expect("Height should not be at its minimum.")
+    }
```

Comment:
I don't disagree with the panics in principle, but most of the functionality implemented for `Height` returns errors vs panics (except the `TryFrom` impl which panics _and_ returns an Error and I'm not sure if that's what we want?) I mostly raise this as we fixed several panics that we turned into errors in the course of the audit so I figured I'd ask

// @teor2345 @arya2 curious on your thoughts too

Reply:
I agree that as a general principle we should handle errors in the caller, rather than panicking for conditions that can happen during normal operation.

But I don't think it's a blocker for this specific PR, so I'd like to deal with the error handling on these new methods and `TryFrom` in a separate ticket or PR.